parser_library/test/context/macro_test.cpp

```diff
@@ -859,3 +859,30 @@ TEST(macro, missing_mend)
 
     EXPECT_TRUE(matches_message_codes(a.diags(), { "E001", "E046" }));
 }
+
+TEST(macro, dynamic_instruction_types)
+{
+    std::string input = R"(
+         MACRO
+         MAC   &OP,&FIRST,&SECOND,&THIRD
+         LCLC  &OPERAND
+&OPERAND SETC  '&FIRST'
+         AIF   ('&SECOND' EQ '').SKIP
+&OPERAND SETC  '&FIRST.,&SECOND.'
+         AIF   ('&THIRD' EQ '').SKIP
+&OPERAND SETC  '&FIRST.,&SECOND.,&THIRD.'
+.SKIP    ANOP
+         &OP   &OPERAND
+         MEND
+
+         MAC SAM64,this_is_comment
+         MAC LA,0,0
+         MAC ARK,0,0,0
+
```

Comment:
```suggestion
         MAC ALIAS,C'somealias'
ALIAS OPSYN SAM64
         MAC ALIAS,this_is_comment
```

I think alias should be explicitly tested as well, since it is handled in a special way.

Also adding opsyn to the mix might be a good idea.